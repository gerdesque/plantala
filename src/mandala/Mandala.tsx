import React, {useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useCanvas } from './useCanvas';
import { IPlant } from '../plant/Plant';

interface IMandalaProps {
  selectedPlants: IPlant[],
}

const useStyles = makeStyles(() => ({
  canvas: {
    width: '100%',
    height: 'auto'
  }
}));

const Mandala = ({ selectedPlants }: IMandalaProps) => {
  const classes = useStyles();
  const { setPlants, canvasRef, canvasWidth, canvasHeight } = useCanvas();

  useEffect(() => {
    setPlants(selectedPlants);
  }, [selectedPlants, setPlants]);

  return (
    <canvas 
      ref={canvasRef}
      className={classes.canvas}
      width={canvasWidth}
      height={canvasHeight}
    />
  );
};

export default Mandala;
