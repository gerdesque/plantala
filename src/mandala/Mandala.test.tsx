import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Mandala from './Mandala';

describe('Mandala', () => {

  let shallow: any;
  let wrapper: any;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Mandala plants={[]} />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a canvas', () => {
    expect(wrapper.find('canvas')).toHaveLength(1);
  });

});
