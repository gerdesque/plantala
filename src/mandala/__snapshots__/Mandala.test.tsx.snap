// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mandala should render correctly 1`] = `
<canvas
  height={499.20000000000005}
  onClick={[Function]}
  width={921.6}
/>
`;
